### PR TITLE
Support decimal to int conversion

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Related to #289

### CODE REVIEW

File: runtime/tests/decimal_conv.c [hunk 1]
```diff
@@ -51,6 +51,44 @@ void testDecToFloat() {
...
+    validateDecToIntNonOverflow("-0.5", 0);
+    validateDecToIntNonOverflow("-1.5", -2);
+    validateDecToIntNonOverflow("-2.5", -2);
+    validateDecToIntNonOverflow("1e2", 100);
+    validateDecToIntNonOverflow("1.5e2", 150);
+    validateDecToIntNonOverflow("-1.5e2", -150);
+    validateDecToIntNonOverflow("1.51e2", 151);
+    validateDecToIntNonOverflow("1.513e2", 151);
+    validateDecToIntNonOverflow("1.515e2", 152);
+    validateDecToIntNonOverflow("9223372036854775807", 9223372036854775807);
```
Comment (jclark) [inline runtime/tests/decimal_conv.c:134]: Should test 9223372036854775807.4 and similar values.

File: runtime/balrt.h [hunk 2]
```diff
@@ -281,6 +281,11 @@ typedef struct {
     PanicCode panicCode;
 } TaggedPtrPanicCode;
 
+typedef struct {
+    int64_t val;
+    bool overflow;
+} DecToIntResult;
```
Comment (jclark) [inline runtime/balrt.h]: This isn't specific to Decimal.  Call it something like `IntWithOverflow`.

File: runtime/balrt.h [hunk 3]
```diff
@@ -281,6 +281,11 @@ typedef struct {
     PanicCode panicCode;
 } TaggedPtrPanicCode;
 
+typedef struct {
+    int64_t val;
```
Comment (jclark) [inline runtime/balrt.h]: Don't abbreviate `val` to `value` when the other field is unabbreviated and longer than `value`.

File: runtime/decimal.c [hunk 4]
```diff
@@ -167,6 +168,35 @@ double _bal_decimal_to_float(TaggedPtr tp) {
     return strtod(dblStr, NULL);
 }
 
+DecToIntResult _bal_decimal_to_int(TaggedPtr tp) {
+    decQuad dQuantize;
+    decQuad dZero;
+    decQuadZero(&dZero);
+    decContext cx;
+    initContext(&cx);
+    decQuadQuantize(&dQuantize, taggedToDecQuad(tp), &dZero, &cx);
```
Comment (jclark) [inline runtime/decimal.c:225]: This is nice!

File: runtime/decimal.c [hunk 5]
```diff
@@ -167,6 +168,35 @@ double _bal_decimal_to_float(TaggedPtr tp) {
...
+    DecToIntResult res;
+    if (cx.status & DEC_Invalid_operation) {
+        // The invalid operation flag is raised 
+        // when 34 digits is not enough to represent quantized decQuad.
+        // This situation can be considered as an overflow scenario.
```
Comment (jclark) [inline runtime/decimal.c]: Are we sure this handles rounding correctly, particularly at the limits?  Say something in the comment about this.

**Review by jclark — CHANGES_REQUESTED** [review]

File: runtime/tests/decimal_conv.c (hunk 1 above)
Comment (KavinduZoysa) [reply runtime/tests/decimal_conv.c:134]: Added more tests

File: runtime/balrt.h (hunk 2 above)
Comment (KavinduZoysa) [reply runtime/balrt.h]: Changed

File: runtime/balrt.h (hunk 3 above)
Comment (KavinduZoysa) [reply runtime/balrt.h]: Changed

File: runtime/decimal.c (hunk 5 above)
Comment (KavinduZoysa) [reply runtime/decimal.c]: @jclark, did you mean handling the rounding when the `DEC_Invalid_operation` is raised?

Comment (KavinduZoysa) [reply runtime/decimal.c]: Updated the comment

**Pull request merged**